Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp

```diff
@@ -104,6 +105,12 @@ void EventDispatcher::initializeConnection(IPC::Connection* connection)
 void EventDispatcher::internalWheelEvent(PageIdentifier pageID, const WebWheelEvent& wheelEvent, RectEdges<bool> rubberBandableEdges, WheelEventOrigin wheelEventOrigin)
 {
     auto processingSteps = OptionSet<WebCore::WheelEventProcessingSteps> { WheelEventProcessingSteps::MainThreadForScrolling, WheelEventProcessingSteps::MainThreadForBlockingDOMEventDispatch };
+
+    if (WebPage* webPage = WebProcess::singleton().webPage(pageID)) {
```

Comment:
You can't call this code here, because we're on the EventDispatcher thread here, and none of the functions are are calling are thread safe.

Reply:
Would this be acceptable?

```
ensureOnMainThread([pageID] {
    if (auto* webPage = WebProcess::singleton().webPage(pageID)) {
        if (auto* keyboardScrollingAnimator = webPage->corePage() ? webPage->corePage()->currentKeyboardScrollingAnimator() : nullptr)
             keyboardScrollingAnimator->stopScrollingImmediately();
    }
});
```